pages/review.tsx

```diff
@@ -82,13 +111,15 @@ const Review = () => {
     }
   };
 
-  function classNames(...classes) {
-    return classes.filter(Boolean).join(' ');
-  }
+  useEffect(() => {
+    const updateContractAddress = async () => {
+      setContractAddress(await signOrderPayment());
+    };
 
-  const [selectedDeliveryMethod, setSelectedDeliveryMethod] = useState(
-    user?.cart?.supportedDeliveryProviders[0],
-  );
+    // if (user?.cart?.paymentInfo?.provider?.type === 'GENERIC') {
+    updateContractAddress();
+    // }
+  }, [user?.cart?.paymentInfo?._id]);
```

Comment:
This should run only if the order has a selected payment provider and that provider is of type GENERIC. currently, it will run no matter if it has a payment provider selected or not.